treeshr/TreeGetNci.c

```diff
@@ -502,7 +502,9 @@ int TreeGetNci(int nid_in, struct nci_itm *nci_itm)
       {
         string = strncpy(malloc(sizeof(NODE_NAME) + 1), node->name,
                          sizeof(NODE_NAME));
-        string[sizeof(NODE_NAME)] = '\0';
+        int length = minInt(strlen(node->name), sizeof(NODE_NAME));
+        string[length] = '\0';
```

Comment:
is this not also part of the next line

Reply:
Yes it is, we solved this twice and forgot to delete one of them, good catch

Reply:
Stephen and I chose the memset() option as it is the more conservative approach (i.e., clears the remainder of the node name memory).

Testing shows that the fix still works fine after removing the extraneous line.   This screenshot shows a new tree displayed in jTraverser2.   Nodes that start with X were added using TCL.   All other nodes were added using jTraverser2.  The tree was written and closed, plus then exited TCL and jTraverser2.   Then re-opened the tree using jTraverser2 and took this screenshot.

![pr2471](https://user-images.githubusercontent.com/98131549/177913821-8597f940-6eed-4121-8369-9708ebaa30bf.jpg)